debug: PuyaPyOptions(output_teal=True, output_source_map=True, output_arc32=False, output_arc56=False, output_ssa_ir=False, output_optimization_ir=False, output_destructured_ir=True, output_memory_ir=False, output_bytecode=True, output_teal_intermediates=False, debug_level=0, optimization_level=2, target_avm_version=10, cli_template_definitions={}, template_vars_prefix='TMPL_', locals_coalescing_strategy=<LocalsCoalescingStrategy.root_operand: 'root_operand'>, _disabled_optimizations=(), paths=['array'], output_awst=False, output_awst_json=False, output_client=False, log_level=<LogLevel.debug: 10>)
info: Found python prefix: <git root>/.venv
debug: Traceback (most recent call last):
  File "<git root>/src/puya/errors.py", line 36, in log_exceptions
    yield
  File "<git root>/src/puyapy/awst_build/context.py", line 137, in log_exceptions
    yield
  File "<git root>/src/puyapy/awst_build/contract.py", line 167, in build
    method_fragment.implementation = method_builder(context)
                                     ^^^^^^^^^^^^^^^^^^^^^^^
  File "<git root>/src/puyapy/awst_build/contract.py", line 321, in <lambda>
    lambda ctx: FunctionASTConverter.convert(
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<git root>/src/puyapy/awst_build/subroutine.py", line 689, in convert
    return cls(
           ^^^^
  File "<git root>/src/puyapy/awst_build/subroutine.py", line 628, in __init__
    translated_body = self.visit_block(func_def.body)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<git root>/src/puyapy/awst_build/subroutine.py", line 789, in visit_block
    translated_stmt = stmt.accept(self)
                      ^^^^^^^^^^^^^^^^^
  File "<git root>/src/puyapy/_vendor/mypy/nodes.py", line 1284, in accept
    return visitor.visit_expression_stmt(self)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<git root>/src/puyapy/awst_build/subroutine.py", line 816, in visit_expression_stmt
    expr_builder = require_instance_builder(stmt.expr.accept(self))
                                            ^^^^^^^^^^^^^^^^^^^^^^
  File "<git root>/src/puyapy/_vendor/mypy/nodes.py", line 1969, in accept
    return visitor.visit_call_expr(self)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<git root>/src/puyapy/awst_build/subroutine.py", line 275, in visit_call_expr
    callee = call.callee.accept(self)
             ^^^^^^^^^^^^^^^^^^^^^^^^
  File "<git root>/src/puyapy/_vendor/mypy/nodes.py", line 1893, in accept
    return visitor.visit_member_expr(self)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<git root>/src/puyapy/awst_build/subroutine.py", line 267, in visit_member_expr
    base = expr.expr.accept(self)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "<git root>/src/puyapy/_vendor/mypy/nodes.py", line 1969, in accept
    return visitor.visit_call_expr(self)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<git root>/src/puyapy/awst_build/subroutine.py", line 280, in visit_call_expr
    return callee.call(
           ^^^^^^^^^^^^
  File "<git root>/src/puyapy/awst_build/eb/transaction/inner_params.py", line 51, in call
    transaction_fields.update(_map_itxn_args(arg_names, args))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<git root>/src/puyapy/awst_build/eb/transaction/inner_params.py", line 142, in _map_itxn_args
    transaction_fields[params.field] = params.validate_and_convert(arg).resolve()
                                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<git root>/src/puyapy/awst_build/eb/transaction/itxn_args.py", line 64, in validate_and_convert
    [self._validate_and_convert_item(item) for item in eb.iterate_static()],
     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<git root>/src/puyapy/awst_build/eb/transaction/itxn_args.py", line 92, in _validate_and_convert_item
    return BytesExpressionBuilder(item.to_bytes(item.source_location))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<git root>/src/puyapy/awst_build/eb/bytes.py", line 161, in __init__
    super().__init__(pytypes.BytesType, expr)
  File "<git root>/src/puyapy/awst_build/eb/_base.py", line 92, in __init__
    raise InternalError(
puya.errors.InternalError: invalid expression wtype 'stack_array<arc4.static_array<arc4.uint8, 32>>' for Python type 'algopy.Bytes'
array/unused_opt_allowed.py:33:23 critical: invalid expression wtype 'stack_array<arc4.static_array<arc4.uint8, 32>>' for Python type 'algopy.Bytes'
                                                       app_args=(address_arr,),
                                                                 ^~~~~~~~~~~